diff --git a/antigen.zsh b/antigen.zsh
index 63d3a3b..8f4f47a 100644
--- a/antigen.zsh
+++ b/antigen.zsh
@@ -277,7 +277,10 @@ antigen-revert () {
         # implementation once tests are ready.
         local script_loc="$(ls "$location" | grep '\.plugin\.zsh$' | head -n1)"
 
-        if [[ -f $location/$script_loc ]]; then
+        if [[ -f $location ]]; then
+            source "$location"
+
+        elif [[ -f $location/$script_loc ]]; then
             # If we have a `*.plugin.zsh`, source it.
             source "$location/$script_loc"
 
